K8s Orion: Release notes (#2411)
RedisLabs/redislabs-docs: content/kubernetes/release-notes/k8s-6-4-2-3.md

---
Title: Redis Enterprise for Kubernetes release notes 6.4.2-3 (March 2023)
linktitle: 6.4.2-3 (March 2023)
description: 
weight: 57
alwaysopen: false
categories: ["Platforms"]
aliases: [ 
    /kubernetes/release-notes/k8s-6-4-2-3.md,
    /kubernetes/release-notes/k8s-6-4-2-3/,   ]
---
## Overview

> The Redis Enterprise K8s 6.4.2-3 release supports the Redis Enterprise Software 6.4.2 and contains new features and feature improvements.

Reviewer: ```suggestion
The Redis Enterprise K8s 6.4.2-3 release supports Redis Enterprise Software 6.4.2 and contains new features and feature improvements.
```

The key features, bug fixes, and known limitations are described below.

## Images

* **Redis Enterprise**: `redislabs/redis:6.4.2-43`
* **Operator**: `redislabs/operator:6.4.2-43`
* **Services Rigger**: `redislabs/k8s-controller:6.4.2-43`

### OpenShift images

**OLM operator version**: `v6.4.2-3`

* **Redis Enterprise**: `registry.connect.redhat.com/redislabs/redis-enterprise:6.4.2-43.rhel8-openshift`
    (or `redislabs/redis-enterprise:6.4.2-43.rhel7-openshift` if upgrading from RHEL 7)
* **Operator**: `registry.connect.redhat.com/redislabs/redis-enterprise-operator:6.4.2-3`
* **Services Rigger**: `registry.connect.redhat.com/redislabs/services-manager:6.4.2-3`

## Upgrade to 6.4.2-3

This release uses a new file for the `ValidatingWebhookConfiguration`. To use the 6.4.2-3 release, delete the old file `redb-admission.yaml` and apply the new file `webhook.yaml`.

## New features

> * Cluster level LDAP configuration support in REC (RED-83533)

Reviewer: ```suggestion
* Cluster-level LDAP configuration support in REC (RED-83533)
```

* Public preview support for declarative Active-Active configuration (RED-86470)(RED-86470)

## Feature improvements

* New labeling method for multi-namespace REDB (RED-83102)
* Operator deployment uses built-in SCC in OpenShift (RED-90524)
* Support for Redis Enterprise 6.4.2-30
* Support added for K8s 1.26, EKS 1.24, AKS 1.25, Rancher 2.7/1.23-24
* Improved log collector display and error handling

## Bug fixes

* Deployment manifests referenced non-existent redis-enterprise-admission ServiceAccount (RED-90469)
* CVE-2022-1996 (RED-93026)
* CVE-2022-28948 (RED-93027)
* CVE-2021-44716 & CVE-2022-27664 (RED-93028)
* CVE-2018-20225

## API changes

* Fields added to REC custom resource:
  * `managedAPIs`
  * `ldapClientCertificateSecretName`
  * `LDAPAuthenticationQuery`
  * `LDAPAuthorizationQuery`
  * `LDAPQuery`
  * `LDAPServer`
  * `LDAPSpec`
* New custom resource definitions for preview features:
  * Redis Enterprise remote cluster (RERC)
  * Redis Enterprise Active-Active database (REAADB)

## Compatibility notes

### Deprecated features

> Support for Gesher, the advanced admission control configuration, is deprecated.

Reviewer 2: Another deprecation: the RedisEnterpriseCluster v1alpha1 API version:

"The **app.redislabs.com/v1alpha1** API version of the `RedisEnterpriseCluster` (REC) custom resource is deprecated and will be removed in a subsequent release. Customers should switch to use the **app.redislabs.com/v1** API version instead."

### Supported distributions

> Below is a table showing supported distributions at the time of this release. See [Supported Kubernetes distributions]({{< relref "/kubernetes/reference/supported_k8s_distributions.md" >}}) for the current list of supported distributions.

Reviewer: ```suggestion
The following table shows supported distributions at the time of this release. You can also find this list in [Supported Kubernetes distributions]({{< relref "/kubernetes/reference/supported_k8s_distributions.md" >}}).
```

| Amazon EKS              |            | supported  | supported  | supported* |            |            |
| Azure AKS               |            |            | deprecated | supported  | supported* |            |
| Google GKE              |            | supported  | supported  | supported  | supported  |            |
| Rancher 2.6             | deprecated | supported  | supported  | supported  |            |            |
| Rancher 2.7             |            |            | supported* | supported* |            |            |
| VMware TKG 1.6          |            | supported  | supported  |            |            |            |
| **OpenShift version**   | **4.8**    | **4.9**    | **4.10**   | **4.11**   |            |            |
|                         |            | deprecated | supported  | supported  |            |            |
| **VMware TKGI version** | **1.12**   | **1.13**   | **1.14**   | **1.15**   |            |            |
|                         |            | deprecated | supported  | supported  |            |            |

\* Support added in this release

## Known limitations

* Long cluster names cause routes to be rejected  (RED-25871)

>   A cluster name longer than 20 characters will result in a rejected route configuration because the host part of the domain name will exceed 63 characters. The workaround is to limit cluster name to 20 characters or less.

Reviewer: ```suggestion
  A cluster name longer than 20 characters will result in a rejected route configuration because the host part of the domain name will exceed 63 characters. The workaround is to limit the cluster name to 20 characters or fewer.
```

* Cluster CR (REC) errors are not reported after invalid updates (RED-25542)
  
  A cluster CR specification error is not reported if two or more invalid CR resources are updated in sequence.

* An unreachable cluster has status running (RED-32805)
  

>   When a cluster is in an unreachable state, the state is still `running` instead of being reported as an error.

Reviewer: ```suggestion
  When a cluster is in unreachable state, the state remains `running` instead of triggering an error.
```

* Readiness probe incorrect on failures (RED-39300)

  STS Readiness probe does not mark a node as "not ready" when running `rladmin status` on node failure.

* Internal DNS and Kubernetes DNS may have conflicts (RED-37462)

  DNS conflicts are possible between the cluster `mdns_server` and the K8s DNS. This only impacts DNS resolution from within cluster nodes for Kubernetes DNS names.

* 5.4.10 negatively impacts 5.4.6 (RED-37233)

  Kubernetes-based 5.4.10 deployments seem to negatively impact existing 5.4.6 deployments that share a Kubernetes cluster.

* Node CPU usage is reported instead of pod CPU usage (RED-36884)

>   In Kubernetes, the node CPU usage we report on is the usage of the Kubernetes worker node hosting the REC pod.

Reviewer: ```suggestion
  In Kubernetes, the reported node CPU usage is the usage of the Kubernetes worker node hosting the REC pod.
```

* Clusters must be named "rec" in OLM-based deployments (RED-39825)

>   In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) are bound to a specific service account name (i.e., "rec"). The workaround is to name the cluster "rec".

Reviewer: ```suggestion
  In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) are bound to a specific service account name (namely, "rec"). The workaround is to name the cluster "rec".
```

* REC clusters fail to start on Kubernetes clusters with unsynchronized clocks (RED-47254)

  When REC clusters are deployed on Kubernetes clusters with unsynchronized clocks, the REC cluster does not start correctly. The fix is to use NTP to synchronize the underlying K8s nodes.

* Deleting an OpenShift project with an REC deployed may hang (RED-47192)

>   When a REC cluster is deployed in a project (namespace) and has REDB resources, the REDB resources must be deleted first before the REC can be deleted. As such, until the REDB resources are deleted, the project deletion will hang. The fix is to delete the REDB resources first and the REC second. Afterwards, you may delete the project.

Reviewer: ```suggestion
  When an REC cluster is deployed in a project (namespace) and has REDB resources, the REDB resources must be deleted first before the REC can be deleted. Therefore, until the REDB resources are deleted, the project deletion will hang. The fix is to delete the REDB resources first and the REC second. Then, you can delete the project.
```

* Hashicorp Vault integration - no support for Gesher (RED-55080)

  There is no workaround at this time.

* REC might report error states on initial startup (RED-61707)

  There is no workaround at this time except to ignore the errors.

* PVC size issues when using decimal value in spec (RED-62132)

  The workaround for this issue is to make sure you use integer values for the PVC size.

* Following old revision of quick start guide causes issues creating an REDB due to unrecognized memory field name (RED-69515)

>   The workaround is to use the newer (current) revision of the [quick start document]({{<relref "/kubernetes/deployment/quick-start.md">}}) available online.

Reviewer: ```suggestion
  The workaround is to use the newer (current) revision of the [quick start]({{<relref "/kubernetes/deployment/quick-start.md">}}).
```

> * `autoUpgrade` set to true by operator might cause unexpected bdb upgrades when `redisUpgradePolicy` is set to true (RED-72351)

Reviewer: ```suggestion
* `autoUpgrade` set to `true` can cause unexpected bdb upgrades when `redisUpgradePolicy` is set to `true` (RED-72351)
```

## Upgrade to 6.4.2-3

> This release uses a new file for the `ValidatingWebhookConfiguration`. To use the 6.4.2-3 release, delete the old file and apply the new file.

Reviewer 2: Maybe phrase this around webhook resources and not around files, something along these lines:

```suggestion
This release uses a new `ValidatingWebhookConfiguration` resource that replaces the old webhook resource. To use the 6.4.2-3 release, delete the old webhook resource and apply the new file.
```
